android/src/main/kotlin/io/customer/customer_io/CustomerIoPlugin.kt

```diff
@@ -1,35 +1,114 @@
-    channel.setMethodCallHandler(null)
-  }
+
+    private fun initialize(call: MethodCall, result: Result) {
+        try {
+            val application: Application = context.applicationContext as Application
+            val configData = call.arguments as? Map<String, Any> ?: emptyMap()
+            val siteId = configData.getString(Keys.Environment.SITE_ID)
+            val apiKey = configData.getString(Keys.Environment.API_KEY)
+            val region = configData.getProperty<String>(
+                Keys.Environment.REGION
+            )?.takeIfNotBlank().toRegion()
+
+            CustomerIO.Builder(
+                siteId = siteId,
+                apiKey = apiKey,
+                region = region,
+                appContext = application,
+            ).apply {
+                setClient(client = getUserAgentClient(packageConfig = configData))
+                setupConfig(configData)
+            }.build()
+            logger.info("Customer.io instance initialized successfully")
+            result.success(true)
+        } catch (e: Exception) {
+            logger.error("Failed to initialize Customer.io instance from app, ${e.message}")
+            result.error("FlutterSegmentException", e.localizedMessage, null);
+        }
+    }
+
+    private fun getUserAgentClient(packageConfig: Map<String, Any?>?): Client {
+        val sourceSDKVersion = packageConfig?.getProperty<String>(
+            Keys.PackageConfig.SOURCE_SDK_VERSION
+        )?.takeIfNotBlank() ?: packageConfig?.getProperty<String>(
+            Keys.PackageConfig.SOURCE_SDK_VERSION_COMPAT
+        )?.takeIfNotBlank() ?: "n/a"
+        // TODO: change it to flutter
+        return Client.ReactNative(sdkVersion = sourceSDKVersion)
+    }
+
+    private fun CustomerIO.Builder.setupConfig(config: Map<String, Any?>?): CustomerIO.Builder {
+        if (config == null) return this
+
+        val logLevel = config.getProperty<String>(Keys.Config.LOG_LEVEL).toCIOLogLevel()
+        setLogLevel(level = logLevel)
+        config.getProperty<String>(Keys.Config.TRACKING_API_URL)?.takeIfNotBlank()?.let { value ->
+            setTrackingApiURL(value)
+        }
+        config.getProperty<Boolean>(Keys.Config.AUTO_TRACK_DEVICE_ATTRIBUTES)?.let { value ->
+            autoTrackDeviceAttributes(shouldTrackDeviceAttributes = value)
+        }
+        config.getProperty<Double>(Keys.Config.BACKGROUND_QUEUE_MIN_NUMBER_OF_TASKS)?.let { value ->
+            setBackgroundQueueMinNumberOfTasks(backgroundQueueMinNumberOfTasks = value.toInt())
+        }
+        config.getProperty<Double>(Keys.Config.BACKGROUND_QUEUE_SECONDS_DELAY)?.let { value ->
+            setBackgroundQueueSecondsDelay(backgroundQueueSecondsDelay = value)
+        }
+        return this
+    }
+
+    override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
+        flutterCommunicationChannel.setMethodCallHandler(null)
```

Comment:
That's a good point, i will look more into it.

Reply:
Checked the documentation, and since the context is a private member its lifecycle is bound to the `plugin` lifecycle. 

And in this method, it will release all the resources held by this plugin.

Reply:
Great, thanks for confirming and sharing the updates 👍🏻